docker/mips64el-linux-musl-gcc.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+# this linker wrapper works around the missing soft-fp routine __trunctfsf2
+# this affects rust versions with compiler-builtins <= 0.1.77,
+# which has not yet been merged into stable. this requires the `-lgcc`
+# linker flag to provide the missing builtin.
+# https://github.com/rust-lang/compiler-builtins/pull/483
```

Comment:
0.1.78 is present in 1.65

Reply:
This can be applied then to both mips64 and mips64le then.

Reply:
Since this logic would now be present in 3 files, but i don't really want to add it to `lib.sh` for full rebuilds, should i just make `rust.sh` which would have `rustc_version` and `rustc_minor`? Something like:

```bash
#!/bin/bash

rustc_version() {
    rustc -Vv | grep '^release:' | cut -d ':' -f2
}

rustc_minor() {
    rustc_version | cut -d '.' -f2
}
```

And then just `rustc_minor` in `aarch64-linux-musl-gcc.sh`, `mips64-linux-musl-gcc.sh`, and `mips64el-linux-musl-gcc.sh` after sourcing `rustc.sh`?

Reply:
we could have cross inform about the rustc version also. An env var like `_RUSTC_MINOR_VERSION` or similar. Technically we should also report the major version, but there are no plans for a rust 2.0 :)

Reply:
We prefix every other one with `CROSS` so `CROSS_RUSTC_MAJOR_VERSION`, `CROSS_RUSTC_MINOR_VERSION`, and `CROSS_RUSTC_PATCH_VERSION` could work nicely. That way, we have no conflicts.

Reply:
Does the latest commit fix this? I've added `rustc_version` to `DockerOptions`, and changed `docker_envvars` to take the `DockerOptions`, and then export the version info via:

```rust
    let (major, minor, patch) = match options.rustc_version.as_ref() {
        Some(version) => (version.major, version.minor, version.patch),
        // no toolchain version available, always provide older
        // compiler available. this isn't a major issue because
        // linking will libgcc will not include symbols found in
        // the builtins.
        None => (1, 0, 0),
    };
    docker.args(["-e", &format!("CROSS_RUSTC_MAJOR_VERSION={}", major)]);
    docker.args(["-e", &format!("CROSS_RUSTC_MINOR_VERSION={}", minor)]);
    docker.args(["-e", &format!("CROSS_RUSTC_PATCH_VERSION={}", patch)]);
```

This also fixes an issue if the `rustc` version doesn't provide the version info, which might have caused a bug in `aarch64-unknown-linux-musl`.